controllers/etcdcluster_controller_test.go

```diff
@@ -498,6 +542,78 @@ func (s *controllerSuite) testClusterController(t *testing.T) {
 			assert.Equal(t, *etcdCluster.Spec.PodTemplate.Resources, r1.Spec.Template.Spec.Containers[0].Resources)
 		})
 	})
+
+	t.Run("StatusVersion", func(t *testing.T) {
+		etcdAPI := &WrapperEtcdAPI{wrapped: &AlwaysFailEtcdAPI{}}
+
+		teardownFunc, namespace := s.setupTest(t, etcdAPI)
+		defer teardownFunc()
+
+		etcdCluster := test.ExampleEtcdCluster(namespace)
+
+		err := s.k8sClient.Create(s.ctx, etcdCluster)
+		require.NoError(t, err, "failed to create EtcdCluster resource")
+
+		etcdCluster.Default()
+
+		etcdClusterKey, err := client.ObjectKeyFromObject(etcdCluster)
+		require.NoError(t, err)
+
+		t.Run("ReportsEmptyVersion", func(t *testing.T) {
+			etcdAPI.Wrap(&AlwaysFailEtcdAPI{})
+
+			err := s.triggerReconcile(etcdCluster)
+			require.NoError(t, err)
+
+			err = try.Eventually(func() error {
+				var fetchedCluster etcdv1alpha1.EtcdCluster
+				err := s.k8sClient.Get(s.ctx, etcdClusterKey, &fetchedCluster)
+				require.NoError(t, err)
+
+				const expectedVersion = ""
+				actualVersion := fetchedCluster.Status.ClusterVersion
+				if expectedVersion != actualVersion {
+					return fmt.Errorf(
+						"unexpected Status.ClusterVersion. Wanted: %s, Got: %s",
+						expectedVersion, actualVersion,
+					)
+				}
+				return nil
+			}, time.Second*5, time.Millisecond*500)
+			require.NoError(t, err)
+		})
+
+		t.Run("ReportsExpectedVersion", func(t *testing.T) {
+			expectedVersion := semver.Must(semver.NewVersion("3.2.0"))
+			staticAPI := fakeEtcdForEtcdCluster(*etcdCluster)
+			staticAPI.ClusterVersion = "3.2.0"
+			etcdAPI.Wrap(staticAPI)
+
+			err := s.triggerReconcile(etcdCluster)
+			require.NoError(t, err)
+
+			err = try.Eventually(func() error {
+
+				var fetchedCluster etcdv1alpha1.EtcdCluster
+				err := s.k8sClient.Get(s.ctx, etcdClusterKey, &fetchedCluster)
+				require.NoError(t, err)
+
+				if len(fetchedCluster.Status.ClusterVersion) == 0 {
```

Comment:
`fetchedCluster.Status.ClusterVersion` is a string right? Why are we using `len(...) == 0` over `... == ""`? Both work of course, but at a casual scan of the code you might think it's a slice.

Reply:
Done.